### ISSUE TITLE
Missing declarations and invalid code

### ISSUE THREAD

**Issue by birkelund** [go-swagger/go-swagger#1062]
## Problem statement
go-swagger generates server with missing declarations and client with invalid code. 

```
$ swagger generate client -f swagger.json
$ go build ./...
client/test/get_test_parameters.go:126: cannot use valuesTestarray (type [][]int32) as type []string in argument to swag.JoinByFormat

$ swagger generate server -f swagger.json
$ go run cmd/example-swagger-interface-server/main.go
restapi/operations/test/get_test_parameters.go:82: undefined: i
restapi/operations/test/get_test_parameters.go:88: undefined: testarrayI
```

## Swagger specification
https://gist.github.com/kbj/b6046cda56e50e027f1f089d13084e1c

## Steps to reproduce
`swagger generate client -f swagger.json`
`swagger generate server -f swagger.json`

## Environment
swagger version: dev (220a68f31437616b53c032a5f82384450f82cf4e)
go version: go1.8.3 linux/amd64
OS: Debian GNU/Linux 9 (testing)

**Comment by kbj** [comment]
I tried to create as simple a `swagger.json` as possible replicating the issue, but it still seems to behave a bit differently from what I originally encountered while generating from a real swagger spec. The spec in question is the [EVE-Online ESI](https://esi.tech.ccp.is/latest/) (which validates correctly). Generating a server from that spec causes the index variable `i` to be generated, but it is missing the `connectionsI` variable. Remember to call `swagger flatten` beforehand to try this (see #984). The issue is in the `/routes/` endpoint.

For the client, the type clash remains between `[][]int32` and `[]string`. I looked into the code and it looks like the template can't generate a "recursive formatter" to convert the int32's to strings.
